packages/contracts-rfq/contracts/FastBridgeV2.sol

```diff
@@ -173,15 +173,15 @@ contract FastBridgeV2 is Admin, IFastBridgeV2 {
             rebate
         );
     }
-
+    
     /// @inheritdoc IFastBridge
     function prove(bytes memory request, bytes32 destTxHash) external {
-        prove(request, destTxHash, msg.sender);
+        bytes32 transactionId = keccak256(request);
+        prove(transactionId, destTxHash, msg.sender);
     }
 
     /// @inheritdoc IFastBridgeV2
-    function prove(bytes memory request, bytes32 destTxHash, address relayer) public onlyRole(RELAYER_ROLE) {
-        bytes32 transactionId = keccak256(request);
+    function prove(bytes32 transactionId, bytes32 destTxHash, address relayer) public onlyRole(RELAYER_ROLE) {
```

Comment:
It seems that we don't currently have coverage for this new function, which explains why tests didn't result in a compilation error. Let's throw in a few cases, could just mirror the existing tests - but focusing on `relayer != msg.sender` cases for consistent coverage.